pkg/k8shandler/collection.go

```diff
@@ -98,7 +98,7 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection(proxyConfi
 
 		printUpdateMessage := true
 		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			if !reflect.DeepEqual(fluentdStatus, cluster.Status.Collection.Logs.FluentdStatus) {
+			if !compareFluentdCollectorStatus(fluentdStatus, cluster.Status.Collection.Logs.FluentdStatus) {
```

Comment:
Is it necessary to compare status? Why is status just not the current state of the collector?

Reply:
if conditions have changed from the last time we processed we need to make sure that we are updating it... if nothing has changed we don't want to bother making the Status update call